pkg/provider/kubernetes/gateway/kubernetes_test.go

```diff
@@ -945,14 +946,78 @@ func TestHostRule(t *testing.T) {
 			},
 			expectedRule: "",
 		},
+		{
+			desc: "Several Host and wildcard",
+			routeSpec: v1alpha1.HTTPRouteSpec{
+				Hostnames: []v1alpha1.Hostname{
+					"*.bar.foo",
+					"bar.foo",
+					"foo.foo",
+				},
+			},
+			expectedRule: "(Host(`bar.foo`, `foo.foo`) || HostRegexp(`{subdomain:[a-zA-Z0-9-]+}.bar.foo`))",
+		},
+		{
+			desc: "Host with wildcard",
+			routeSpec: v1alpha1.HTTPRouteSpec{
+				Hostnames: []v1alpha1.Hostname{
+					"*.bar.foo",
+				},
+			},
+			expectedRule: "HostRegexp(`{subdomain:[a-zA-Z0-9-]+}.bar.foo`)",
+		},
+		{
+			desc: "Alone wildcard",
+			routeSpec: v1alpha1.HTTPRouteSpec{
+				Hostnames: []v1alpha1.Hostname{
+					"foo.foo",
+					"*",
```

Comment:
I think technically this is allowed too, though I care much less about supporting it as you could just not specify a host rule at all. And technically an empty string is `*` too.